src/bindings/jna/libelektra4j/src/main/java/org/libelektra/PluginLoader.java

```diff
@@ -0,0 +1,40 @@
+package org.libelektra;
+
+import org.libelektra.exception.InstallationException;
+import org.libelektra.plugin.Echo;
+import org.libelektra.plugin.NativePlugin;
+import org.libelektra.plugin.PropertiesStorage;
+import org.libelektra.plugin.Return;
+
+/**
+ * This class can be used to load Plugins from Elektra.
+ * It also loads self implemented Java plugins.
```

Comment:
Here I would have the expectation that also my newly implemented Java plugin can be loaded (which obviously is not the case as the loading is hard-coded).

Reply:
I could use reflection but this comes with several drawbacks.
From java 9 on Reflection is not very favorable anymore
Furthermore one would need to force a default constructor so every plugin can be instantiated via reflection.

I will add it to the README that it also has to be added to the PluginLoader implementation.

Reply:
Yes, I also think the readme how to add the plugin would be enough.

What is not so ideal in the current solution that the Java plugins shadow other plugins. If there is also a native "echo" plugin, it would not be possible to load it.

Maybe we drop the idea to "load" Java plugins altogether.

Reply:
> What is not so ideal in the current solution that the Java plugins shadow other plugins. If there is also a native "echo" plugin, it would not be possible to load it.

I agree with you. I separate the methods into a "loadNativePlugin" and "loadJavaPlugin" to make it clear.

Reply:
good idea :+1: